syntax_and_semantics/annotations/user_defined_annotations.md

```diff
@@ -0,0 +1,174 @@
+# User-defined annotations
+
+Users can define their own annotations using the `annotation` keyword, which works similarly to defining a `class` or `struct`.
+
+```crystal
+annotation MyAnnotation; end
+```
+
+The annotation can then be applied to various items, including:
+* Instance and class methods
+* Instance variables
+* Classes, structs, and modules
+
+```crystal
+annotation MyAnnotation; end
+
+@[MyAnnotation]
+def foo
+  "foo"
+end
+
+@[MyAnnotation]
+class Klass
+end
+
+@[MyAnnotation]
+module MyModule
+end
+```
+
+## Reading
+
+Annotations can be read off of a [`TypeNode`](https://crystal-lang.org/api/Crystal/Macros/TypeNode.html), [`Def`](https://crystal-lang.org/api/Crystal/Macros/Def.html), or [`MetaVar`](https://crystal-lang.org/api/Crystal/Macros/MetaVar.html) using the `[]` method.  This method return an [`Annotation`](https://crystal-lang.org/api/master/Crystal/Macros/Annotation.html) object representing the applied annotation of the supplied type.
```

Comment:
Isn't it the `.annotation` method instead of `[]`?

Reply:
`.annotation` returns the `Annotation` type object from the `TypeNode`.  However `[]` is used to read a named/positional value from the annotation.